gluoncv/loss.py

```diff
@@ -133,8 +137,8 @@ def forward(self, cls_pred, box_pred, cls_target, box_target):
             pos_samples = (ct > 0)
             num_pos.append(pos_samples.sum())
         num_pos_all = sum([p.asscalar() for p in num_pos])
-        if num_pos_all < 1:
-            # no positive samples found, return dummy losses
+        if num_pos_all < 1 and self._min_hard_negatives < 1:
+            # no positive samples and no hard negatives, return dummy losses
```

Comment:
I think to avoid step error with no gradients, here we can use:
```python
cls_losses = [nd.sum(cp * 0) for cp in cls_pred]
box_losses = [nd.sum(bp * 0) for bp in box_pred]
sum_losses = [nd.sum(cp * 0) + nd.sum(bp * 0) for cp, bp in zip(cls_pred, box_pred)]
return sum_losses, cls_losses, box_losses
```

Reply:
I agree, this has now been changed in the commit.